test/peaks/yearn/YVaultController.test.js

```diff
@@ -0,0 +1,124 @@
+const assert = require('assert')
+const utils = require('../../utils.js')
+
+const toWei = web3.utils.toWei
+const fromWei = web3.utils.fromWei
+const toBN = web3.utils.toBN
+const MAX = web3.utils.toTwosComplement(-1);
+const n_coins = 4
+
+
+contract('YVaultController', async (accounts) => {
+
+    before(async () => {
+        const artifacts = await utils.getArtifacts()
+        Object.assign(this, artifacts)
+        this.owner = accounts[0]
+
+        this.amounts = [200, 200, 200, 200] // dai, usdc, usdt, tusd
+        this.reserves = this.reserves.slice(0, 3).concat([this.reserves[4]])
+        const tasks = []
+        for (let i = 0; i < n_coins; i++) {
+            this.amounts[i] = toBN(this.amounts[i]).mul(toBN(10 ** this.decimals[i])) // convert amounts
+            tasks.push(this.reserves[i].mint(this.owner, this.amounts[i])) // mint user
+        }
+        await Promise.all(tasks)
+    })
+
+    // Owner tests
+    describe('Only Owner', async () => {
+        it('Owner can add Peak', async () => {
+            let reverted = false
+            try {
+                await this.controller.addPeak(this.yVaultPeak.address, {from: this.owner})
```

Comment:
Since this call is not expected to revert, it's of no use putting in a try-catch block. Even in the `catch` block, I don't think `ERR_PEAK` error is being emitted anywhere.

We will need another test that tries to add a peak from a non-owner account and that is the one that is expected to revert.

For this test, you should add the peak and read `peaks[this.yVaultPeak.address]` to assert that the peak was in fact added.